lib/engine/step/g_18_co/dividend.rb

```diff
@@ -14,17 +14,15 @@ def share_price_change(entity, revenue = 0)
           { share_direction: %i[right up], share_times: [1, 1] }
         end
 
-        def change_share_price(entity, payout)
-          tfm = @game.mines_total(entity)
-
-          # Only pay mines if a train produce revenue regardless of withhold or pay
-          if tfm.positive? && (payout[:corporation].positive? || payout[:per_share].positive?)
-            @game.bank.spend(tfm, entity)
+        def process_dividend(action)
```

Comment:
Pulled this from @michaeljb 's suggestions as it is a much more appropriate place to execute the action.